# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/interfaces/base/nsIServiceWorkerManager.idl
# Commit: f567e48e5957
# Full Hash: f567e48e5957604f7c561642bfc1ffa12ba1d299
# Author: Luca Greco <lgreco@mozilla.com>
# Date: 2020-07-03 15:35:32
# Regressor Bug: 1609920
# File Overlap Count: 2
# Description:
#   Bug 1609920 - part 1: Allow the WebExtension Framework to register a moz-extension service worker. r=dom-workers-and-storage-reviewers,asuth,mixedpuppy
#   
#   - Adds the new about:config pref "extensions.backgroundServiceWorker.enabled" (currently defaults to false).
#   - Adds the background.service_worker property to the manifest JSON schema definition
#   - Locks background.service_worker manifest property behind the new preference
# ==============================================================================

diff -r 2b880e8d9143 -r f567e48e5957 dom/interfaces/base/nsIServiceWorkerManager.idl
--- a/dom/interfaces/base/nsIServiceWorkerManager.idl	Fri Jul 03 11:27:56 2020 +0000
+++ b/dom/interfaces/base/nsIServiceWorkerManager.idl	Fri Jul 03 10:14:24 2020 +0000
@@ -135,6 +135,15 @@
                           in AString aScriptURL);
 
   /**
+   * Register an extension background service worker for a given
+   * extension principal and return a promise that resolves to the
+   * nsIServiceWorkerRegistrationInfo (or rejects if there was one
+   * already registered).
+   */
+  [implicit_jscontext]
+  Promise registerForAddonPrincipal(in nsIPrincipal aPrincipal);
+
+  /**
    * Unregister an existing ServiceWorker registration for `aScope`.
    * It keeps aCallback alive until the operation is concluded.
    */